### PR TITLE
fix: log details of cert attachment use case

### PR DESCRIPTION
**Pull request by jcosentino11**
**Issue #, if available:**

**Description of changes:**

The goal of the change is to clearly log the results of cert attachment, such that a user is able to debug why attachment may have failed, possibly due to local attachment expiring, for example.

Also added missing test coverage for verifying expired local cert attachment

**Why is this change necessary:**
We recently had a UAT failure likely due to failed attachment verification, due to expiry. This was determined by examining timestamps, so this change would remove any guesswork.

**How was this change tested:**

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by MikeDombo — COMMENTED** [review]
This feels like a lot to change just to get a few more logs. Any room to simplify this?

File: src/main/java/com/aws/greengrass/clientdevices/auth/iot/Thing.java [hunk 1]
```diff
@@ -113,11 +113,7 @@ public Map<String, Instant> getAttachedCertificateIds() {
      * @param certificateId - A certificateId
      */
     public Optional<Instant> certificateLastAttachedOn(String certificateId) {
-        if (!attachedCertificateIds.containsKey(certificateId)) {
-            return Optional.empty();
-        }
-
-        return Optional.of(attachedCertificateIds.get(certificateId));
+        return Optional.ofNullable(attachedCertificateIds.get(certificateId));
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/clientdevices/auth/iot/Thing.java:116]: good call, that was a race before.

File: src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/CreateIoTThingSession.java [hunk 2]
```diff
@@ -64,16 +67,23 @@ public Session apply(CreateSessionDTO dto) throws AuthenticationException {
...
 
-            if (thingAttachedResult) {
+            logger.atDebug()
+                    .kv("thingIsAttached", result.isThingAttachedToCertificate())
+                    .kv("lastAttachedOn", result.getLastAttached())
+                    .kv("attachmentExpiration", result.getAttachmentExpiration())
+                    .kv("source", result.getVerificationSource())
+                    .log("Attachment verification result");
+
+            if (result.isThingAttachedToCertificate()) {
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/CreateIoTThingSession.java:80]: check for expiration here too, not just that it is attached?

Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/CreateIoTThingSession.java:80]: ah yeah that was a miss, fixing

Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/CreateIoTThingSession.java:80]: spoke too soon, it's covered here: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/pull/238/files#diff-d564d0cc489527df3742905afa00509b6cfb6ebdb07597c35138150a4b2c1a45R56

probably needs a wording

File: src/main/java/com/aws/greengrass/clientdevices/auth/iot/Thing.java (hunk 1 above)
Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/clientdevices/auth/iot/Thing.java:116]: :)

File: src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/CreateIoTThingSession.java (hunk 2 above)
Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/clientdevices/auth/iot/usecases/CreateIoTThingSession.java]: changed wording from `thingAttachedToCertificate` to `thingHasValidAttachmentToCertificate`,  hopefully more clear

**Review by MikeDombo — APPROVED** [review]

**Comment by MikeDombo** [comment]
Updated title. This is a fix, not a feature. Feature would mean that we need to bump the minor version and that's definitely no the case for this change.

**Review by robcmann — APPROVED** [review]

**Pull request merged**